cmd/repo_review.go

```diff
@@ -0,0 +1,154 @@
+package cmd
+
+import (
+	"fmt"
+	"github.com/fatih/color"
+	"github.com/gsoares85/code-guardian/internal/github_internal"
+	"github.com/gsoares85/code-guardian/internal/openai"
+	"github.com/spf13/cobra"
+	"path/filepath"
+	"strings"
+	"time"
+)
+
+var repoReviewCmd = &cobra.Command{
+	Use:   "repo-review [owner] [repo] [flags]",
+	Short: "Analyse an entire github repository",
+	Long: `This command scans all source files recursively in a repository
+           to train the AI about the application and provide:
+           - A summary of what the application does
+           - Key use cases
+           - A high-level code quality review (only critical issues)
+           - A high-level security review (only critical issues)
+           - Key improvement areas`,
+	Args: cobra.ExactArgs(2),
+	Run: func(cmd *cobra.Command, args []string) {
+		owner, repo := args[0], args[1]
+		saveOutput, err := cmd.Flags().GetBool("output")
+		if err != nil {
+			color.Red("Error: %v", err)
+			return
+		}
+
+		color.Blue("\n🔍 Fetching all source code files recursively...\n")
+		files, err := github_internal.GetRepositoryFilesRecursive(owner, repo)
+		if err != nil {
+			color.Red("❌ ERROR: Fetching repository files: %s\n", err)
+			return
+		}
+
+		color.Green("📂 Repository contains %d files\n", len(files))
+
+		sourceCode := fetchAllSourceCode(owner, repo, files)
+		if len(sourceCode) == 0 {
+			color.Red("❌ ERROR: No valid source code found for analysis")
+			return
+		}
+
+		color.Blue("\n🤖 Training AI with full source code...\n")
+		summary, useCases, codeReview, securityReview, improvements := analyzeRepositoryWithAI(sourceCode)
+
+		displayRepoAnalysis(repo, summary, useCases, codeReview, securityReview, improvements)
+
+		if saveOutput {
+			saveRepoAnalysis(repo, owner, summary, useCases, codeReview, securityReview, improvements)
+		}
+	},
+}
+
+func fetchAllSourceCode(owner, repo string, files []string) string {
+	var allCode strings.Builder
+
+	for _, file := range files {
+		if strings.HasSuffix(file, ".md") || strings.Contains(file, "LICENSE") {
+			continue
+		}
+
+		color.Cyan("\n📄 Reading file: %s", file)
+
+		content, err := github_internal.GetFileContent(owner, repo, file)
+		if err != nil {
+			color.Red("❌ ERROR fetching file content: %s\n", err)
+			continue
```

Comment:
1. In the file `cmd/repo_review.go`, within the `fetchAllSourceCode` function, we are ignoring any errors that occurred during the `github_internal.GetFileContent` function call. It would be better to also handle these errors properly instead of simply continuing the execution. It would be better to include some form of exception handling, for example logging it for future usage.